service/history/workflow/mutable_state_impl.go

```diff
@@ -1605,6 +1605,7 @@ func (ms *MutableStateImpl) UpdateActivityInfo(
 
 	ai.Version = incomingActivityInfo.GetVersion()
 	ai.ScheduledTime = incomingActivityInfo.GetScheduledTime()
+	// we don't need to update FirstScheduledTime
```

Comment:
I wonder how reset activity would interact with first scheduled time. Are we going to reset that? Is this going to be done via a history event or applied directly to the state from the RPC? That's an important future detail to keep in mind.

Reply:
yes, when activity is reset - schedule_to_close timer starting stime will also be reseted. It will be applyed directly from the RPC. It was mentined in design document.